sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/ParameterValidationInAsyncShouldBeWrapped.cs

```diff
@@ -90,5 +73,26 @@ public static Task SkipLinesAsync(this TextReader reader, int linesToSkip)
                 throw new Exception("Wrong age");
             }
         }
+
+
+        public static Task SkipLinesAsync(this TextReader reader, int linesToSkip)
+        {
+            if (reader == null)
```

Comment:
If you are going to change this test, I would recommend simplifying it as well - for example, only one argument on the outer method is enough and the loop inside the local function is not necessary.